src/request/sign-swap/SignSwapApi.js

```diff
@@ -145,14 +153,37 @@ class SignSwapApi extends PolygonRequestParserMixin(BitcoinRequestParserMixin(To
                 amount: this.parsePositiveInteger(request.redeem.amount, false, 'redeem.amount'),
             };
         } else if (request.redeem.type === 'EUR') {
+            const settlement = this.parseOasisSettlementInstruction(request.redeem.settlement, 'redeem.settlement');
+            if (
+                settlement.type !== 'sepa'
+                && (CONFIG.NETWORK === Constants.NETWORK.MAIN || settlement.type !== 'mock')
+            ) {
+                throw new Errors.InvalidRequestError('Invalid redeeming settlement type');
+            }
             parsedRequest.redeem = {
                 type: 'EUR',
                 keyPath: this.parsePath(request.redeem.keyPath, 'redeem.keyPath'),
-                settlement: this.parseOasisSettlementInstruction(request.redeem.settlement, 'redeem.settlement'),
+                settlement,
                 amount: this.parsePositiveInteger(request.redeem.amount, false, 'redeem.amount'),
                 fee: this.parsePositiveInteger(request.redeem.fee, true, 'redeem.fee'),
                 bankLabel: this.parseLabel(request.redeem.bankLabel, true, 'redeem.bankLabel'),
             };
+        } else if (request.redeem.type === 'CRC') {
+            const settlement = this.parseOasisSettlementInstruction(request.redeem.settlement, 'redeem.settlement');
+            if (
+                settlement.type !== 'sinpemovil'
+                && (CONFIG.NETWORK === Constants.NETWORK.MAIN || settlement.type !== 'mock')
+            ) {
+                throw new Errors.InvalidRequestError('Invalid redeeming settlement type');
+            }
+            parsedRequest.redeem = {
+                type: 'CRC',
+                keyPath: this.parsePath(request.redeem.keyPath, 'redeem.keyPath'),
+                settlement,
+                amount: this.parsePositiveInteger(request.redeem.amount, false, 'redeem.amount'),
+                fee: this.parsePositiveInteger(request.redeem.fee, true, 'redeem.fee'),
+                recipientLabel: this.parseLabel(request.redeem.recipientLabel, true, 'redeem.recipientLabel'),
+            };
```

Comment:
Here too, I think this can be reasonably well be merged with the `request.redeem.type === 'EUR'` case.